k-distribution/include/ocaml/prelude.ml

```diff
@@ -835,6 +835,9 @@ struct
   let hook_min c _ _ _ _ = match c with
       [Int a], [Int b] -> [Int (Z.min a b)]
     | _ -> raise Not_implemented
+  let hook_log2 c _ _ _ _ = match c with
```

Comment:
similarly here, it seems you could be faster by using number of limbs, limb size, and computing the bit length of the most significant limb.

Reply:
The performance is more than good enough with this as is.

Reply:
I could maybe do what you suggest but it would be more fragile, so I'm inclined to leave it.